python/cudnn/grouped_gemm/grouped_gemm_dglu/moe_blockscaled_grouped_gemm_dglu_rubin.py

```
@@ -0,0 +1,4286 @@
+                    #
+                    if cutlass.const_expr(self.store_d_directly):
+                        self.epilog_sync_barrier.arrive_and_wait()
+                        d_idx_mn = (mma_tile_coord_mnl[0], mma_tile_coord_mnl[1])
+                        d_epilogue_subtile = (
+                            cute.make_layout(128),
+                            cute.make_layout(self.mma_tiler[1] * 2),
+                        )
+                        gD_sub_loop = cute.local_tile(
+                            real_d, d_epilogue_subtile, (None, None, None)
+                        )
+                        tCgD_mnl_loop = thr_copy_t2r.partition_D(gD_sub_loop)
```

Ruff (0.16.0)

[error] Undefined name `thr_copy_t2r`

(F821)